### PR TITLE
Add handling for the Pro plugin

### PR DESCRIPTION
**Pull request by kienstra**
#### Changes
* On entering a valid subscription key, this downloads the Pro plugin, not the free one.

#### Testing instructions
1. If you have GCB Pro in `wp-content/plugins/`, rename it to `genesis-custom-blocks-pro-renamed/` so this can install GCB Pro from the endpoint
1. Ensure there is a valid Block Lab Pro subscription key saved
1. Go to `/wp-admin` > Migrate 
1. Enter a valid Genesis Pro subscription key
1. Expected: here-after-submitting
1. Continue through the migration flow to actually installing GCB
1. Expected: After clicking 'Install and activate,' this should actually install GCB Pro from the endpoint, and it should work
1. The version of GCB Pro it downloads from the endpoint isn't the final one that'll be released, but it's pretty close

### CODE REVIEW

File: php/admin/migration/class-api.php [hunk 1]
```diff
@@ -20,90 +20,15 @@
...
-		add_action( 'rest_api_init', [ $this, 'register_route_update_subscription_key' ] );
 		add_action( 'rest_api_init', [ $this, 'register_route_install_gcb' ] );
 		add_action( 'rest_api_init', [ $this, 'register_route_migrate_post_content' ] );
 		add_action( 'rest_api_init', [ $this, 'register_route_migrate_post_type' ] );
 	}
 
-	/**
-	 * Registers a route to migrate the post type.
-	 */
-	public function register_route_update_subscription_key() {
```
Comment (kienstra) [inline php/admin/migration/class-api.php:65]: Just moved to the class `Subscription_Api`

**Comment by kienstra** [comment]
Hi @lukecarbis,
Could you please review this? 

On entering a Genesis Pro key, it validates it, and gets the download link for GCB Pro.

Then, it uses that download link to install that plugin.

Also, should I remove this?

genesis-pro-copy

Thanks, Luke!

**Comment by lukecarbis** [comment]
Awesome, Ryan! Couple of things:

1. There should be a spinner when the user clicks "Save" to indicate that the save process is happening.
2. The Save button should have a `:hover` and `:active` state, like other buttons. Something like:
```css
color: var(--color-purple-800);
background-color: var(--color-purple-300);
```
3. Let's remove the coupon code generation process. We're not doing that anymore.
4. Let's change the text etc. as follows:
Screen Shot 2020-08-27 at 12 27 29 pm

> Since you're a Block Lab Pro customer, we've already emailed you regarding setting up a WP Engine account with a free Pro subscription.
>
> To migrate and maintain your Block Lab Pro feature set with Genesis Custom Blocks, you will need your Genesis Pro subscription key.
>
> - Already have got it? Enter the subscription key below to continue migrating.
> - Don’t have one yet? Please opt-in using the link below.
>
> [Opt-in for Genesis Pro] (This may take up to 3 working days)

The button should link to: https://forms.gle/26u7NDRUp2A9i2aF8

5. Can we hide the text input if the key is validated successfully? Just to make it clearer that the "Next Step"
is available and needs to be clicked

File: js/migration/components/steps/get-genesis-pro.js [hunk 2]
```diff
@@ -79,11 +75,13 @@ const GetGenesisPro = ( { goToNext, isStepActive, isStepComplete, stepIndex } )
 			method: 'POST',
 			data: { subscriptionKey },
 		} ).then( () => {
-			setSubmissionMessage( __( 'Thanks, the key was saved.', 'block-lab' ) );
+			setSubmissionMessage( __( 'Thanks, the key is valid.', 'block-lab' ) );
 			setKeySubmittedSuccessfully( true );
-		} ).catch( () => {
-			setSubmissionMessage( __( 'There was an error saving the key.', 'block-lab' ) );
+		} ).catch( ( error ) => {
+			const errorMessage = error.message ? error.message : __( 'There was an error in validating the key.', 'block-lab' );
```
Comment (lukecarbis) [inline js/migration/components/steps/get-genesis-pro.js]:
```suggestion
			const errorMessage = error.message ? error.message : __( 'There was an error validating the key.', 'block-lab' );
```

File: js/migration/components/steps/test/get-genesis-pro.js [hunk 3]
```diff
@@ -48,7 +45,7 @@ test( 'get Genesis Pro migration step', async () => {
 	await waitFor( () =>
 		user.click( getByText( 'Save' ) )
 	);
-	getByText( 'Thanks, the key was saved.' );
+	getByText( 'Thanks, the key is valid.' );
```
Comment (lukecarbis) [inline js/migration/components/steps/test/get-genesis-pro.js]:
```suggestion
	getByText( 'Thanks! Your key is valid, and has been saved.' );
```

**Review by lukecarbis — CHANGES_REQUESTED** [review]

**Comment by lukecarbis** [comment]
BTW, the I did get a download link in the transient, but the download didn't work. The download link looked like: `https://s3.amazonaws.com/{domain}/genesis-custom-blocks-pro/genesis-custom-blocks-pro.latest.zip?AWSAccessKeyId={key}&Signature={signature}%3D&Expires={timestamp}`

**Comment by kienstra** [comment]
>BTW, the I did get a download link in the transient, but the download didn't work. The download link looked like...

Ah, I saw that too. I think the `download_link` was a dummy link, the actual plugin isn't deployed.

**Comment by kienstra** [comment]
Thanks, really good suggestions! If it's alright, I'll apply them tomorrow.

**Comment by kienstra** [comment]
Before submitting the sub key:

here-working-days

After submitting the key:

here-after-submitting

**Comment by kienstra** [comment]
Hi @lukecarbis,
Could you please review this? It applies all of your ideas and copy from https://github.com/getblocklab/block-lab/pull/688#issuecomment-681306113.

Also, the [top comment](https://github.com/getblocklab/block-lab/pull/688#issue-473726400) is up to date.

I'll open a PR from `develop` to `add/full-migration`, resolving merge conflicts. After that's merged, this PR will probably have merge conflicts that I'll resolve.

**Review by lukecarbis — APPROVED** [review]
Perfect @kienstra!! Nice work. Looks fantastic.

The download worked nicely too, now that the Pro version is available via that endpoint.

💜

**Comment by lukecarbis** [comment]
Goodbye Block Lab… 😭

**Comment by kienstra** [comment]
@lukecarbis, thanks for testing this!

**Pull request merged**